src/core/storage/sframe_data/sframe.cpp

```diff
@@ -539,6 +539,12 @@ sframe::iterator sframe::get_output_iterator(size_t segmentid) {
   return sframe::iterator(
       [=](const std::vector<flexible_type>& val)->void{
         // check that all the types match up
+        if (UNLIKELY(val.size() != ctypes.size())) {
+          std::stringstream ss;
+          ss << "Write to sframe with row size mismatch. "
```

Comment:
I think this would be clearer:

```
          ss << "Can not write to SFrame, got the wrong number of columns. "
	     << "Expected: " << ctypes.size() << " columns. Got: " << val.size() << " columns.";
```

Reply:
will add more detailed info!